examples/js/loaders/OBJLoader.js

```diff
@@ -318,70 +318,80 @@ THREE.OBJLoader.prototype = {
 
 			},
 
-			addFace: function ( a, b, c, d, ua, ub, uc, ud, na, nb, nc, nd ) {
+			addFace: function ( vs, us, ns ) {
 
 				var vLen = this.vertices.length;
+				var ivs = [];
 
-				var ia = this.parseVertexIndex( a, vLen );
-				var ib = this.parseVertexIndex( b, vLen );
-				var ic = this.parseVertexIndex( c, vLen );
-				var id;
+				for ( var j = 0; j < vs.length; j++ ) {
 
-				if ( d === undefined ) {
+				  ivs.push( this.parseVertexIndex( vs[ j ], vLen ) )
 
-					this.addVertex( ia, ib, ic );
+				}
 
-				} else {
+				if ( vs.length > 3 ) {
+
+					for ( var j = 2; j < ivs.length; j ++ ) {
+
+						this.addVertex( ivs[ 0 ], ivs[ j - 1 ], ivs[ j ] );
```

Comment:
> So I believe this only works for convex faces ? And can potentially produce very high aspect ratio triangles ?
Yes. The former OBJLoader connect neighboring pair of vertices with last vertex.
> Is three ok with zero area triangles (when the three vertices are colinear) ?
I think the model editor need to take care of this. (broken model)